### PR TITLE
chores/version_migration

### PR DESCRIPTION
**Pull request by fhuaulme**
Prototype for on-the-fly MongoDB document migration

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/daikon_pull-request_master/87/) [:x:](https://ci-common.datapwn.com/job/daikon_pull-request_master/87/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/88/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/88/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/89/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/89/)

File: daikon-spring/daikon-migration-core/README.MD [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+Then use `MigrationRegister`:
+```java
+class Main {
+    public static void main(String... args) {
+        ObjectNode jsonObject = ... // omitted
+        Function<ObjectNode, String> version = ... // a way to extract version from JSON object
+        A instanceOfA = new A();
+        
+        MigrationRegister.register(A.class, Rule_200.class); // Important since A no longer has inner class.
```
Comment (sgandon) [inline daikon-spring/daikon-migration-core/README.MD]: do you think the registration of the migration rule could not be done from the @MigrationRule definition (line 77) 
simething like
`@MigrationRule(version = "2.0.0", A.class)`

**Comment by sgandon** [comment]
nice !

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/daikon_pull-request_master/41/) [:x:](https://ci-common.datapwn.com/job/daikon_pull-request_master/41/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/55/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/55/)

**Comment by sgandon** [comment]
just a test to see if LucidChart can be displayed in Github
https://www.lucidchart.com/invitations/accept/cdcfe741-f83a-4361-971e-df6a1d6fba97

**Pull request closed without merging**